### ISSUE TITLE
Dead code warning for main() function.

### ISSUE THREAD

**Issue by fkjogu** [AtomLinter/linter-rust#32]
This code
```rust
fn main() {

}
```
raises a warning 
```
function is never used: `main`, #[warn(dead_code)] on by default at line 1 col 1
```
during linting. Not to be expected ;).

Using Atom 1.0.2, with linter-rust 0.2.5, linter 1.2.4 on Debian Jessie.

**Comment by cyberDrake** [comment]
Same issue here, using Atom 1.0.3, linter-rust 0.2.5, linter 1.2.4, OSX 10.10.4

**Comment by dgriffen** [comment]
Also having the same issue, linter-rust is currently unusable for me because almost all functions get labeled as unused. A normal cargo build does not produce this error. Issue appears to be from commit 8731c7247816da8ca255d4a0d358c4a5b7114fda where the cargo command was changed to test instead of build.

**Comment by dgriffen** [comment]
After looking at some discussion [here](https://github.com/rust-lang/rust/issues/12327) it seems the best solution would be to have a setting for whether or not source code is compiled with a test configuration. The warnings can be useful to determine if functions are missing a test case, but this should not be the default behavior.
